src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -920,8 +920,23 @@ void FixAndProcessAdoNetEvents(TraceEvent data)
                         return;
                     StartStopActivity creator = GetCurrentStartStopActivity(thread, data);
 
-                    Debug.Assert(data.payloadNames[1] == "dataSource");
-                    var extraStartInfo = data.PayloadValue(1) as string;
+                    string extraStartInfo = null;
+                    if (3 < data.payloadNames.Length)
```

Comment:
:bulb: Placing constants on the right side of expressions is typically more readable in C#

Reply:
I use a convention that all comparisons go from small to big.

Reply:
`std::less` :trollface: